src/main/java/ch/njol/skript/effects/EffConnect.java

```diff
@@ -29,65 +10,104 @@
+	"When transferring players using an IP, the transfer will not complete if the `accepts-transfers` option isn't enabled in `server.properties` for the server specified.",
+	"If the port is not provided, it will default to `25565`."
+})
+@Examples({
+	"connect all players to proxy server \"hub\"",
+	"transfer player to server \"my.server.com\"",
+	"transfer player to server \"localhost\" on port 25566"
+})
+@Since("2.3, INSERT VERSION (transfer)")
 public class EffConnect extends Effect {
 
 	public static final String BUNGEE_CHANNEL = "BungeeCord";
 	public static final String GET_SERVERS_CHANNEL = "GetServers";
 	public static final String CONNECT_CHANNEL = "Connect";
+	private static final boolean TRANSFER_METHOD_EXISTS = Skript.methodExists(Player.class, "transfer", String.class, int.class);
 
 	static {
 		Skript.registerEffect(EffConnect.class,
-				"connect %players% to [server] %string%",
-				"send %players% to server %string%"
+				"(transfer|send|connect) %players% to (proxy|bungeecord) server %string%",
+				"(transfer|send|connect) %players% to server %string% [on port %-number%]"
 		);
 	}
 
-	@SuppressWarnings("null")
 	private Expression<Player> players;
-
-	@SuppressWarnings("null")
 	private Expression<String> server;
+	private Expression<Number> port;
+	private boolean transfer;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		players = (Expression<Player>) exprs[0];
+		server = (Expression<String>) exprs[1];
+		transfer = matchedPattern == 1;
+
+		if (transfer) {
+			port = exprs.length > 2 ? (Expression<Number>) exprs[2] : null;
+			if (!TRANSFER_METHOD_EXISTS) {
+				Skript.error("Transferring players via IP is not available on this version.");
+				return false;
+			}
+		}
+		return true;
+	}
 
 	@Override
-	protected void execute(Event e) {
-		String server = this.server.getSingle(e);
-		Player[] players = this.players.stream(e)
+	protected void execute(Event event) {
+		String server = this.server.getSingle(event);
+		Player[] players = this.players.stream(event)
 			.filter(Player::isOnline)
 			.toArray(Player[]::new);
+
 		if (server == null || players.length == 0)
 			return;
 
-		// the message channel is case sensitive so let's fix that
-		Utils.sendPluginMessage(players[0], BUNGEE_CHANNEL, r -> GET_SERVERS_CHANNEL.equals(r.readUTF()), GET_SERVERS_CHANNEL)
-			.thenAccept(response -> {
-				// for loop isn't as pretty a stream, but will be faster with tons of servers
-				for (String validServer : response.readUTF().split(", ")) {
-					if (validServer.equalsIgnoreCase(server)) {
-						for (Player player : players)
-							Utils.sendPluginMessage(player, BUNGEE_CHANNEL, CONNECT_CHANNEL, validServer);
-						break;
-					}
+		if (transfer) {
+			int port = 25565; // default port
+			if (this.port != null) {
+				Number portNum = this.port.getSingle(event);
+				if (portNum != null) {
```

Comment:
if the port expression was provided but had a null value, we should probably return rather than defaulting to 25565